code/datums/elements/weapon_description.dm

```diff
@@ -0,0 +1,185 @@
+/**
+ *
+ * The purpose of this element is to widely provide the ability to examine an object and determine its damage, with the ability to add
+ * additional notes or information based on type or other factors
+ *
+ */
+/datum/element/weapon_description
+	/// Flavor text crimes used in build_weapon_text()
+	var/list/crimes = list("Assaults", "Third Degree Murders", "Robberies", "Terrorist Attacks", "Different Felonies", "Felinies", "Counts of Tax Evasion", "Mutinies")
+	var/list/victims = list("a human", "a moth", "a felinid", "a lizard", "a particularly resilient slime", "a syndicate Aagent", "a clown", "a mime", "a mortal foe", "an innocent bystander")
+
+/datum/element/weapon_description/Attach(datum/target)
+	. = ..()
+	if(!isitem(target)) /// Do not attach this to anything that isn't an item
+		return ELEMENT_INCOMPATIBLE
+	RegisterSignal(target, COMSIG_PARENT_EXAMINE, .proc/warning_label)
+	RegisterSignal(target, COMSIG_TOPIC, .proc/topic_handler)
+
+/datum/element/weapon_description/Detach(datum/target)
```

Comment:
If you want your detach to be called, you need to add `element_flags = ELEMENT_DETACH`.